File: generator/src/test/resources/swift/expected/RequestHeaderParamsTest/test-basic-header-parameter-generation.output.swift
import Sunday

public class API {

  public let requestFactory: RequestFactory
  public let defaultContentTypes: [MediaType]
  public let defaultAcceptTypes: [MediaType]

  public init(
    requestFactory: RequestFactory,
    defaultContentTypes: [MediaType] = [],
    defaultAcceptTypes: [MediaType] = [.json]
  ) {
    self.requestFactory = requestFactory
    self.defaultContentTypes = defaultContentTypes
    self.defaultAcceptTypes = defaultAcceptTypes
  }

  public func fetchTest(
    obj: Test,
    strReq: String,
    int: Int = 5
  ) async throws -> Test {
    return try await self.requestFactory.result(
      method: .get,
      pathTemplate: "/tests",
      pathParameters: nil,
      queryParameters: nil,
      body: Empty.none,
      contentTypes: nil,
      acceptTypes: self.defaultAcceptTypes,
      headers: [
        "obj": obj,
        "str-req": strReq,
        "int": int
      ]
    )
  }

  public func deleteTest() async throws {
    return try await self.requestFactory.result(
      method: .delete,
      pathTemplate: "/tests",
      pathParameters: nil,
      queryParameters: nil,
      body: Empty.none,
      contentTypes: nil,
      acceptTypes: self.defaultAcceptTypes,
      headers: nil
    )
  }

}

File: generator/src/test/resources/kotlin/expected/jaxrs/RequestHeaderParamsTest/test-basic-header-parameter-generation.output.kt
package io.test.service

import io.test.Test
import javax.ws.rs.Consumes
import javax.ws.rs.DELETE
import javax.ws.rs.DefaultValue
import javax.ws.rs.GET
import javax.ws.rs.HeaderParam
import javax.ws.rs.Path
import javax.ws.rs.Produces
import javax.ws.rs.core.Response
import kotlin.Int
import kotlin.String

@Produces(value = ["application/json"])
@Consumes(value = ["application/json"])
public interface API {
  @GET
  @Path(value = "/tests")
  public fun fetchTest(
    @HeaderParam(value = "obj") obj: Test,
    @HeaderParam(value = "str-req") strReq: String,
    @HeaderParam(value = "int") @DefaultValue(value = "5") int: Int,
  ): Response

  @DELETE
  @Path(value = "/tests")
  public fun deleteTest(): Response
}
